src/modes/draw_line_string.js

```diff
@@ -60,17 +45,28 @@ module.exports = function(ctx) {
       ctx.ui.setActiveButton(Constants.types.LINE);
       this.on('mousemove', CommonSelectors.true, handleMouseMove);
       this.on('click', CommonSelectors.true, handleClick);
-      this.on('keyup', CommonSelectors.isEscapeKey, stopDrawingAndRemove);
-      this.on('keyup', CommonSelectors.isEnterKey, finish);
+      this.on('keyup', CommonSelectors.isEscapeKey, () => {
+        ctx.store.delete([line.id], { silent: true });
+        ctx.events.changeMode(Constants.modes.SIMPLE_SELECT);
+      });
+      this.on('keyup', CommonSelectors.isEnterKey, () => {
+        ctx.events.changeMode(Constants.modes.SIMPLE_SELECT, { featureIds: [line.id] });
+      });
     },
 
     stop() {
       doubleClickZoom.enable(ctx);
       ctx.ui.setActiveButton();
 
-      // If it's invalid, just destroy the thing
-      if (!line.isValid()) {
+      line.removeCoordinate(`${currentVertexPosition}`);
+      if (line.isValid() && ctx.store.get(line.id) !== undefined) {
+        ctx.map.fire(Constants.events.CREATE, {
+          features: [line.toGeoJSON()]
+        });
+      }
+      else if (ctx.store.get(line.id) !== undefined) {
         ctx.store.delete([line.id], { silent: true });
+        ctx.events.changeMode(Constants.modes.SIMPLE_SELECT, {}, { silent: true });
```

Comment:
I think the repetition of these two lines in 3 places justifies something like the `finish` function, right?